### ISSUE TITLE
[question] demangleAll + regexp + "_Z"

### ISSUE THREAD

**Issue by harsszegi** [emscripten-core/emscripten#18578]
Hi,

I have an issue with WASM loading in Angular. During WASM loading there are certain expected errors to happen (like around Errno handling). In Angular my callstack contains "_ZoneDelegate", which means that the binding code matches to it, because of the following function:

function demangleAll(text) {
  var regex = /\b_Z[\w\d_]+/g;
  return text.replace(regex, function(x) {
    var y = demangle(x);
    return x === y ? x : y + " [" + x + "]";
  });
}

This has a disastrous effect as __cxa_demangle will not be found in my module and loading the WASM fails because of this:

assert(__cxa_demangle_func);

Any hint how to overcome this?
Thanks,

**Comment by sbc100** [comment]
It looks like that `assert(__cxa_demangle_func)` is wrapped in `$if DEMANGLE_SUPPORT`.. so the compiler seems to think that demangle support is enabled.    Are you setting `-sDEMANGLE_SUPPORT` on the command line?

**Comment by harsszegi** [comment]
Yeah I’m setting this, because I’d like to have demangled names in exceptions coming from C++ land, though in this specific case the exception came from JS land. Although I have to state that I have no clue why it didn’t find the cxa_demangle symbol in the image (regardless that it wasn’t a c++ function that failed). I’ll try to see what is tge cxa_demangle signature in my wasm

**Comment by sbc100** [comment]
I created #18579 which should at least make it impossible for `__cxa_demangle` to be missing here.

**Comment by harsszegi** [comment]
Lame question. Doing wasm2wat on my file it states that cxa_demangle is within the WASM

  (func $__cxa_demangle (type 8) (param i32 i32 i32 i32) (result i32)

However Module appears to be missing it runtime. How can that be?
Thanks,
